File: rust/src/models/select_option.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct SelectOption {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub label: Option<String>,

    pub value: String,
}

impl SelectOption {
    pub fn new(value: String) -> Self {
        Self { label: None, value }
    }
}

File: typescript/src/models/selectOption.ts
// this file is @generated

export interface SelectOption {
  label?: string | null;

  value: string;
}

export const SelectOptionSerializer = {
  _fromJsonObject(object: any): SelectOption {
    return {
      label: object["label"],
      value: object["value"],
    };
  },

  _toJsonObject(self: SelectOption): any {
    return {
      label: self.label,
      value: self.value,
    };
  },
};
